### DISCUSSION TITLE
React Diagram API

### DISCUSSION THREAD

**Discussion by gbl-frey** [likec4/likec4#1095] [Ideas]
Would it be possible to open up react diagram by exposing API to enable control of currently zoomed/active node from outside of component?

I think this functionality is already build in as dynamic views support zooming on different nodes, and maybe it's possible to expose hooks that would enable this from the outside?

**Comment by davydkov** [comment]
Hey @gbl-frey 

Nice suggestion, I've created #1199.

Do you have any ideas what else in Diagram API may be useful (besides "zooming")?

**Comment by davydkov** [comment]
https://likec4.dev/tooling/react/#hooks
